cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/table/field/FieldFactory.java

```diff
@@ -0,0 +1,111 @@
+   * @param fieldName field name
+   * @param value the value of the field
+   * @return a field with Integer as value
+   * @throws InvalidFieldException if the field name is not in schema or the type does not match or the field is a key
+   * but provided value is null.
+   */
+  public Field<Integer> createIntField(String fieldName, @Nullable Integer value) throws InvalidFieldException {
+   return validateAndCreateField(fieldName, FieldType.Type.INTEGER, tableSchema.getType(fieldName), value);
+  }
+
+  /**
+   * Create a {@link Field} which has value of Long type.
+   *
+   * @param fieldName field name
+   * @param value the value of the field
+   * @return a field with Long as value
+   * @throws InvalidFieldException if the field name is not in schema or the type does not match or the field is a key
+   * but provided value is null.
+   */
+  public Field<Long> createLongField(String fieldName, @Nullable Long value) throws InvalidFieldException {
+    return validateAndCreateField(fieldName, FieldType.Type.LONG, tableSchema.getType(fieldName), value);
+  }
+
+  /**
+   * Create a {@link Field} which has value of String type.
+   *
+   * @param fieldName field name
+   * @param value the value of the field
+   * @return a field with String as value
+   * @throws InvalidFieldException if the field name is not in schema or the type does not match or the field is a key
+   * but provided value is null.
+   */
+  public Field<String> createStringField(String fieldName, @Nullable String value) throws InvalidFieldException {
+    return validateAndCreateField(fieldName, FieldType.Type.STRING, tableSchema.getType(fieldName), value);
+  }
+
+  /**
+   * Create a {@link Field} which has value of Double type.
+   *
+   * @param fieldName field name
+   * @param value the value of the field
+   * @return a field with Double as value
+   * @throws InvalidFieldException if the field name is not in schema or the type does not match or the field is a key
+   * but provided value is null.
+   */
+  public Field<Double> createDoubleField(String fieldName, @Nullable Double value) throws InvalidFieldException {
+    return validateAndCreateField(fieldName, FieldType.Type.DOUBLE, tableSchema.getType(fieldName), value);
+  }
+
+  /**
+   * Create a {@link Field} which has value of Float type.
+   *
+   * @param fieldName field name
+   * @param value the value of the field
+   * @return a field with Float as value
+   * @throws InvalidFieldException if the field name is not in schema or the type does not match or the field is a key
+   * but provided value is null.
+   */
+  public Field<Float> createFloatField(String fieldName, @Nullable Float value) throws InvalidFieldException {
+    return validateAndCreateField(fieldName, FieldType.Type.FLOAT, tableSchema.getType(fieldName), value);
+  }
+
+  private <T> Field<T> validateAndCreateField(String fieldName, FieldType.Type expected,
+                                          @Nullable FieldType.Type actual,
```

Comment:
this method has access to the schema, so the actual type does not need to be passed in.